Shared/Samples/Browse OGC API feature service/BrowseOGCAPIFeatureServiceView.swift

```diff
@@ -0,0 +1,256 @@
+                        color: .blue,
+                        size: 5
+                    )
+                )
+            }
+            if geometryType == Multipoint.self {
+                renderer = SimpleRenderer(
+                    symbol: SimpleMarkerSymbol(
+                        style: .circle,
+                        color: .blue,
+                        size: 5
+                    )
+                )
+            }
+            if geometryType == Polyline.self {
+                renderer = SimpleRenderer(
+                    symbol: SimpleLineSymbol(
+                        style: .solid,
+                        color: .blue,
+                        width: 1
+                    )
+                )
+            }
+            if geometryType == Polygon.self {
+                renderer = SimpleRenderer(
+                    symbol: SimpleFillSymbol(
+                        style: .solid,
+                        color: .blue,
+                        outline: nil
+                    )
+                )
+            }
+            return renderer
+        }
+        
+        /// Create and load the OGC API features service from a URL.
+        /// - Parameter url: The URL of the OGC service.
+        /// - Returns: Returns a `OCGFeatureService` that has been loaded and initialized.
+        private func makeService(url: URL) async throws -> OGCFeatureService {
+            let service = OGCFeatureService(url: url)
+            try await service.load()
+            if service.loadStatus == .loaded,
+               let serviceInfo = service.serviceInfo {
+                featureCollectionInfos = serviceInfo.featureCollectionInfos
+                layerNames = featureCollectionInfos.map(\.title)
+            }
+            return service
+        }
+        
+        /// Loads OGC service for a URL so that it can be rendered on the map.
+        /// - Parameters:
+        ///   - mapProxy: Allows access to `MapView`
+        ///   - url: The URL of the OGC service.
+        func loadOGCFeatureData(mapProxy: MapViewProxy, url: URL?) async throws {
+            service = try await makeService(
+                url: url ?? .defaultServiceURL
+            )
+            try await displayLayer(
+                with: featureCollectionInfos[0],
+                proxy: mapProxy
+            )
+        }
+        
+        /// Updates the selected info property for the users selection.
+        /// - Parameter selection: String with the name of the selected layer to display.
+        func update(for selection: String) {
+            selectedInfo = featureCollectionInfos.first(where: {
+                $0.title == selection
+            })
+        }
+        
+        /// Load and display a feature layer from the OGC feature collection table.
+        /// - Parameter info: The `OGCFeatureCollectionInfo` selected by user.
+        func displayLayer(with info: OGCFeatureCollectionInfo, proxy: MapViewProxy) async throws {
```

Comment:
I don't like that the map proxy is being passed through two function to get here, is there a way to move proxy logic up the chain?